main.tf

```diff
@@ -1,71 +1,93 @@
 # -------------------------------------------------------------------------------
 # Resources
 # -------------------------------------------------------------------------------
-data "aws_availability_zones" "available" {}
+data "aws_availability_zones" "available" {
+}
 
 resource "random_string" "suffix" {
   length  = 8
   special = "false"
 }
 
 resource "aws_rds_cluster" "main" {
+  depends_on                   = ["aws_db_subnet_group.main"]
   cluster_identifier           = "${var.name_prefix}-cluster"
-  database_name                = "${var.database_name}"
-  master_username              = "${var.username}"
-  master_password              = "${var.password}"
-  port                         = "${var.port}"
-  engine                       = "${var.engine}"
-  engine_version               = "${var.engine_version}"
+  database_name                = var.database_name
+  master_username              = var.username
+  master_password              = var.password
+  port                         = var.port
+  engine                       = var.engine
+  engine_version               = var.engine_version
   backup_retention_period      = 7
   preferred_backup_window      = "02:00-03:00"
   preferred_maintenance_window = "wed:04:00-wed:04:30"
-  snapshot_identifier          = "${var.snapshot_identifier}"
+  snapshot_identifier          = var.snapshot_identifier
   final_snapshot_identifier    = "${var.name_prefix}-final-${random_string.suffix.id}"
-  skip_final_snapshot          = "${var.skip_final_snapshot}"
-  vpc_security_group_ids       = ["${aws_security_group.main.id}"]
+  skip_final_snapshot          = var.skip_final_snapshot
+  vpc_security_group_ids       = [aws_security_group.main.id]
 
-  storage_encrypted = "${var.storage_encrypted}"
-  kms_key_id        = "${var.kms_key_arn}"
+  storage_encrypted = var.storage_encrypted
+  kms_key_id        = var.kms_key_arn
 
   # NOTE: This is duplicated because subnet_group does not return the name.
   db_subnet_group_name = "${var.name_prefix}-subnet-group"
 
-  tags = "${merge(var.tags, map("Name", "${var.name_prefix}-cluster"))}"
+  tags = merge(
+    var.tags,
+    {
+      "Name" = "${var.name_prefix}-cluster"
+    },
+  )
 }
 
 resource "aws_rds_cluster_instance" "main" {
-  count                = "${var.instance_count}"
+  count                = var.instance_count
   identifier           = "${var.name_prefix}-instance-${count.index + 1}"
-  cluster_identifier   = "${aws_rds_cluster.main.id}"
-  instance_class       = "${var.instance_type}"
-  engine               = "${var.engine}"
-  engine_version       = "${var.engine_version}"
-  db_subnet_group_name = "${aws_db_subnet_group.main.name}"
-  publicly_accessible  = "${var.publicly_accessible}"
+  cluster_identifier   = aws_rds_cluster.main.id
+  instance_class       = var.instance_type
+  engine               = var.engine
+  engine_version       = var.engine_version
+  db_subnet_group_name = aws_db_subnet_group.main.name
```

Comment:
If we can reference the name we should also fix this line: https://github.com/telia-oss/terraform-aws-rds-cluster/blob/master/main.tf#L30-L31

Reply:
Lets solve this in another PR and not block this.